core-rust/mesh-api-server/src/mesh_api/conversions/addressing.rs

```diff
@@ -0,0 +1,161 @@
+        return Err(ExtractionError::InvalidCurrency {
+            message: format!("currency {} is not fungible type", currency.symbol),
+        });
+    }
+
+    let divisibility: FungibleResourceManagerDivisibilityFieldSubstate =
+        read_optional_main_field_substate(
+            database,
+            resource_node_id,
+            &FungibleResourceManagerField::Divisibility.into(),
+        )
+        .ok_or_else(|| ExtractionError::InvalidCurrency {
+            message: format!("currency {} not found", currency.symbol),
+        })?;
+    let divisibility = *divisibility.payload().as_unique_version() as i32;
+
+    if divisibility != currency.decimals {
+        return Err(ExtractionError::InvalidCurrency {
+            message: format!(
+                "currency {} decimals mismatch, specified: {}, current: {}",
+                &currency.symbol, &currency.decimals, divisibility
+            ),
+        });
+    }
+    Ok(resource_address)
+}
+
+pub fn to_api_resource_address(
+    context: &MappingContext,
+    resource_address: &ResourceAddress,
+) -> Result<String, MappingError> {
+    to_api_entity_address(context, resource_address.as_node_id())
+}
+
+pub fn to_api_entity_address(
+    context: &MappingContext,
+    node_id: &NodeId,
+) -> Result<String, MappingError> {
+    context
+        .address_encoder
+        .encode(node_id.as_ref())
+        .map_err(|err| MappingError::InvalidEntityAddress { encode_error: err })
+}
+
+pub fn to_mesh_api_account_from_address(
+    mapping_context: &MappingContext,
+    address: impl AsRef<NodeId>,
+) -> Result<models::AccountIdentifier, MappingError> {
+    let node_id: &NodeId = address.as_ref();
+    let address = to_api_entity_address(mapping_context, node_id)?;
+
+    if !node_id.is_global_account() {
+        return Err(MappingError::InvalidAccount {
+            message: format!("address {} is not an account", address),
+        });
+    }
+
+    // See https://docs.cdp.coinbase.com/mesh/docs/models#accountidentifier for field
+    // definitions
+    Ok(models::AccountIdentifier {
+        address,
+        sub_account: None,
+        metadata: None,
+    })
+}
+
+pub fn extract_account(
+    extraction_context: &ExtractionContext,
+    account_identifier: &models::AccountIdentifier,
+) -> Result<ComponentAddress, ExtractionError> {
+    if account_identifier.sub_account.is_some() {
+        return Err(ExtractionError::InvalidAccount {
+            message: format!("Sub accounts not supported"),
+        });
+    }
+    let component_address =
+        extract_component_address(extraction_context, &account_identifier.address)?;
+
+    if component_address.as_node_id().is_global_account() {
+        Ok(component_address)
+    } else {
+        Err(ExtractionError::InvalidAccount {
+            message: format!("address {} is not an account", account_identifier.address),
+        })
+    }
+}
+
+pub fn to_mesh_api_currency_from_resource_address(
```

Comment:
VERY MINOR / Optional: I might be tempted to move this to a `currency.rs` file

Reply:
Will do